### DISCUSSION TITLE
How to change error message in file validator sub rules?

### DISCUSSION THREAD

**Discussion by DannyloAlves** [adonisjs/core#2823] [Help] [answered]
I'm with problem that i can change standard rule, but i can not change sub rules in file validator.
I'm stuck to days in that, please somebody can help me?

`const imageSchema = schema.create({
            cover_image: schema.file({
                size: '2mb',
                extnames: ['jpg', 'png'],
            }),
        })`

`const messages = {
            'cover_image.required': 'Need sent a image.',
            'cover_image.size': 'The image can't be larger that 2mb.',
            'cover_image.extnames': 'Only images jpg or png are allowed.',
        }`

`const imgload = await request.validate({ schema: imageSchema, messages: messages })`

"cover_image.required" work and custom error message is displayed.
But "'cover_image.size" and "cover_image.extnames" not work and standard message is displayed in place.

Somebody know how make work?

**Comment by arthurcvm** [comment, marked as answer]
Hi! Please send the console.log of the size and extnames errors, and after test this:

```ts
const messages = {
 'cover_image.required': 'Need sent a image.', 
 'cover_image.file.size': 'The image can't be larger that 2mb.', 
 'cover_image.file.extnames': 'Only images jpg or png are allowed.' }
```

**Reply by DannyloAlves** [reply]
Hi friend. 
I tried `'cover_image.file.size':` and work.
But `'cover_image.file.extnames':` not work, however i tried `'cover_image.file.extname':` and work!

Thank you so much for help me, graces your help i achieved.

**Reply by thetutlage** [reply]
Fixed in docs too https://docs.adonisjs.com/guides/validator/custom-messages#file

**Reply by arthurcvm** [reply]
Yes @DannyloAlves, look the docs of @thetutlage sent.
